api/src/services/meta.ts

```diff
@@ -38,16 +38,24 @@ export class MetaService {
 	async totalCount(collection: string): Promise<number> {
 		const dbQuery = this.knex(collection).count('*', { as: 'count' }).first();
 
-		if (this.accountability?.admin !== true) {
-			const permissionsRecord = this.accountability?.permissions?.find((permission) => {
-				return permission.action === 'read' && permission.collection === collection;
-			});
-
-			if (!permissionsRecord) throw new ForbiddenError();
-
-			const permissions = permissionsRecord.permissions ?? {};
-
-			applyFilter(this.knex, this.schema, dbQuery, permissions, collection, {});
+		if (this.accountability) {
+			await validateAccess(
+				{
+					accountability: this.accountability,
+					action: 'read',
+					collection,
+				},
+				{
+					schema: this.schema,
+					knex: this.knex,
+				},
+			);
+
+			// const policies = await fetchPolicies(this.accessService, this.accountability);
+			// const permissions = await fetchPermissions(this.permissionsService, 'read', policies, collection ? [collection] : undefined);
+
+			// TODO replace with applyCases (TBD)
+			// applyFilter(this.knex, this.schema, dbQuery, permissions, collection, {});
```

Comment:
When we bring this back, we need to make sure to include `&& this.accountability.admin === true` in the if above, otherwise all fields will be filtered out.

Reply:
Yeah no kidding right